scan/lib/scan/runner.rb

```diff
@@ -80,8 +84,16 @@ def handle_results(tests_exit_status)
     end
 
     def test_results
-      return "" unless Scan.cache[:temp_junit_report]
-      File.read(Scan.cache[:temp_junit_report])
+      temp_junit_report = Scan.cache[:temp_junit_report]
+      return File.read(temp_junit_report) if temp_junit_report && File.file?(temp_junit_report)
```

Comment:
Also regarding point 2, https://github.com/fastlane/fastlane/issues/9285 points out that when using `output_style: raw`, we return early before appending the `--report --output` parts (intentionally). [source](https://github.com/fastlane/fastlane/blob/2.35.0/scan/lib/scan/test_command_generator.rb#L81). This means that when using `output_style: raw`, we have no choice to use the pre 2.34.0 method of generating a test results table using `cat` instead of my optimization.